src/logic.js

```diff
@@ -1,14 +1,22 @@
 define([
-    'vellum/util',
     'xpath',
     'xpathmodels',
     'underscore'
 ], function (
-    util,
     xpath,
     xpathmodels,
     _
 ) {
+    var XPATH_REFERENCES = [
+        "relevantAttr",
+        "calculateAttr",
+        "constraintAttr",
+        "dataParent",
+        "repeat_count",
+        "filter",
+        "defaultValue"
+    ], NO_SELF_REFERENCES = _.without(XPATH_REFERENCES, 'constraintAttr');
```

Comment:
Yeah. I looked through this list and it seemed like none of them should allow you to do that, but missed `constraintAttr`. (Would have probably caused an infinite loop with something like `dataParent`, trying to find it's own parent in the data tree)